### PR TITLE
[0] Development lifecycle improvements 

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Updating env config
Chose babel-plugin-inline-dotenv over react-native-dotenv because this person's rationale seemed quite reasonable
https://stackoverflow.com/questions/57060316/environment-variables-in-babel-config-js

Placed a `.env.example` file in root - duplicate it to create `.env.development` and `.env.production` files. For now, the latter two files will be the same. Fill in your Airtable API key, and the IMG API Key from @JohnathanZhou. 
**Please make sure not to commit your old `environment.js` file after pulling from master**

### Airtable refactoring
- See Aivant's Airtable generator repo on Github https://github.com/aivantg/airtable-schema-generator/
!! Important: I haven't actually changed our Airtable calls to use this yet - I'm still figuring out what I want that to look like. That'll be in the PR after the restructuring PR #24

### Package updates
I resolved all the errors and warnings from `yarn` when running `yarn install`. This will be done automatically from now on.

#### Post-pull/merge/etc: yarn package management

Using package `yarnhook` to watch `package.json` and `yarn.lock` for changes (I think); basically it'll automatically trigger a `yarn install` to make sure your packages are up-to-date.

#### Pre-commit: Prettier and ESLint run before commit
`husky` enables easy usage of git hooks.

As the name suggests, `lint-staged` runs a series of scripts on only staged files

**From now on, contributors  _must_ have ESLint working with the VSCode setup. Why? Because we'll add back the pre-commit `lint-staged` script that runs `eslint --fix` and won't allow you to commit without fixing the errors.**

!! We will not allow you to make any contributions to the codebase without following ESLint now (^:

#### Aside: VSCode auto-save conflicts 
(_whack thing that was happening to me_)

`yarn add --dev eslint-config-prettier eslint-plugin-prettier`

In `.eslintrc.js`: 
```
	plugins: ['prettier'],
	extends: ['plugin:prettier/recommended']
```
** Make sure it's at the end of the 'extends' list, so it can override any existing configs. See https://prettier.io/docs/en/integrating-with-linters.html

In your VSCode `settings.json`, make sure these lines exist
```
	"editor.defaultFormatter": "esbenp.prettier-vscode",
    "[javascript]": {
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
	"editor.formatOnSave": true,
    "diffEditor.ignoreTrimWhitespace": false,
    "editor.codeActionsOnSave": {
        "source.fixAll.eslint": true
    },
```
!! Make sure no other conflicts exist! look for 'format on' 'autofix' 'fix on save', etc
!! Also, note that the `source.fixAll.eslint` attribute can be really annoying. It's up to you if you want to keep it.

### TypeScript migration preparation

Add this to your VSCode `settings.json`:

```
    "javascript.suggest.completeFunctionCalls": true,
    "javascript.implicitProjectConfig.checkJs": true,
```

### PR Template
I've added a template to `.github/pr_template.md`.
I've also updated this page in our Notion: https://www.notion.so/Development-Lifecycle-db58003c8ae5451c92682f5cfe4fe491
I suggest everyone favorite it until it becomes second nature to follow this. It's just good to have good practices, and to leave an electronic trail in general.

### README updates
I created the skeleton for the README. It didn't seem prudent to document too much, as we're in the midst of a huge overhaul (^:

### Asana task(s)
N/A

[//]: # "Add the relevant Asana link(s) here"

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

Adding PR template and flow mods:
- Added husky, lint-staged, and yarn-hook
- https://codeburst.io/continuous-integration-lint-staged-husky-pre-commit-hook-test-setup-47f8172924fc
- https://medium.com/@bartwijnants/using-prettier-and-husky-to-make-your-commits-save-2960f55cd351
- https://github.com/frontsideair/yarnhook (keep node_modules up-to-date)
- https://github.com/okonet/lint-staged
- https://www.npmjs.com/package/husky
- https://git-scm.com/docs/githooks
- PR template taken from Unloop/PP Power; modified according to my feelings
	- Can be found in `.github/pr_template.md

Prettier config for scripts:
- https://github.com/prettier/prettier/issues/6280 

### Related PRs
#24 
TBA: "[2] Airtable API calls refactoring" will link back in when created.

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
[TBD]

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
See "Related PRs" section

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

### Screenshots
No screenshots for now.

CC: @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

**Comment by anniero98** [comment]
Tagging in @ethanlee16 @FranCogMents @aivantg @kyuwonk (as well as #24 ) if any of y'all have time since these are both relatively big architectural changes and wanted to get feedback from you! cc @wangannie

File: lib/common.js [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-import getEnvVars from '../environment';
-
 // BASE is shared across our screens for Airtable API calls
 /**
  * Usage:
  * import BASE from "../lib/common"
  * BASE("Products")...
  * */
+const BASE = new Airtable({
+  apiKey: process.env.AIRTABLE_API_KEY
```
Comment (aivantg) [inline lib/common.js]: When you do decide to integrate the schema generator, note that this api key will need to be renamed to follow the schema generator's rules

File: package.json [hunk 2]
```diff
@@ -44,33 +67,38 @@
...
+    "eslint-plugin-prettier": "^3.1.2",
     "eslint-plugin-react": "^7.16.0",
     "eslint-plugin-react-hooks": "^1.7.0",
     "eslint-plugin-react-native": "^3.8.1",
+    "husky": "^4.2.1",
     "jest-expo": "^35.0.0",
-    "prettier": "^1.18.2"
+    "lint-staged": "^10.0.6",
+    "prettier": "^1.18.2",
+    "yarnhook": "^0.4.3"
```
Comment (aivantg) [inline package.json]: this seems like a very cool package!!

Comment (anniero98) [reply package.json]: It's very cool !! (:

**Pull request merged**